src/commands/env/create/sandbox.ts

```diff
@@ -0,0 +1,258 @@
+ */
+
+import * as fs from 'fs';
+import { Flags } from '@oclif/core';
+import { SfCommand } from '@salesforce/sf-plugins-core';
+import {
+  AuthInfo,
+  Lifecycle,
+  Messages,
+  Org,
+  ResultEvent,
+  SandboxEvents,
+  SandboxProcessObject,
+  SandboxRequest,
+  SandboxUserAuthResponse,
+  SfError,
+  StatusEvent,
+} from '@salesforce/core';
+import { Duration } from '@salesforce/kit';
+import { Ux } from '@salesforce/sf-plugins-core/lib/ux';
+import { getLogSandboxProcessResult, getSandboxProgress, SandboxProgress } from '../../../shared/sandboxReporter';
+import { handleSideEffects, toKeyValuePairs } from '../../../utils';
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/plugin-env', 'create.sandbox');
+const progressBarOptions = {
+  title: 'Sandbox Create',
+  format: '%s | {id} | {status} | {bar} | {percentComplete}%. {remainingWaitTimeHuman}',
+  barCompleteChar: '\u2588',
+  barIncompleteChar: '\u2591',
+  linewrap: true,
+};
+
+export default class CreateSandbox extends SfCommand<SandboxProcessObject> {
+  public static summary = messages.getMessage('summary');
+  public static description = messages.getMessage('description');
+  public static examples = messages.getMessages('examples');
+
+  public static flags = {
+    // needs to change when new flags are available
+    'definition-file': Flags.string({
+      char: 'f',
+      summary: messages.getMessage('flags.definitionFile.summary'),
+    }),
+    'set-default': Flags.boolean({
+      char: 's',
+      summary: messages.getMessage('flags.setDefault.summary'),
+    }),
+    alias: Flags.string({
+      char: 'a',
+      summary: messages.getMessage('flags.alias.summary'),
+    }),
+    wait: Flags.integer({
+      char: 'w',
+      summary: messages.getMessage('flags.wait.summary'),
+      // min: 6,
+      default: 6,
+    }),
+    'def-property': Flags.string({
+      char: 'p',
+      summary: messages.getMessage('flags.defProperty.summary'),
+      multiple: true,
+    }),
+    'target-org': Flags.string({
+      char: 'o',
+      summary: messages.getMessage('flags.targetOrg.summary'),
+    }),
+  };
+
+  protected readonly lifecycleEventNames = ['postorgcreate'];
+  private sandboxAuth?: SandboxUserAuthResponse;
+  private flags!: {
+    'definition-file': string;
+    'set-default': boolean;
+    alias: string;
+    wait: number;
+    json: boolean;
+    'def-property': string[];
+    'target-org': string;
+  };
+  private sandboxDefinitionProperties: Record<string, string>;
+  public async run(): Promise<SandboxProcessObject> {
+    this.flags = (await this.parse(CreateSandbox)).flags;
+    this.debug('Create started with args %s ', this.flags);
+
+    this.validateSandboxFlags();
+    return await this.createSandbox();
+  }
+
+  private validateSandboxFlags(): void {
+    if (this.flags['def-property']) {
+      this.sandboxDefinitionProperties = toKeyValuePairs(this.flags['def-property']);
+    }
```

Comment:
@peternhale in this case we definitely know the error is with the `def-property` flag because that's what we're passing into `toKeyValuePairs`.

The code here is just to explain the desired behavior of stating to the user, in the error message, which flag is causing a problem. I'm open to other implementations that don't require catching and throwing another error.